lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/interfaces.hpp

```diff
@@ -111,4 +96,62 @@ class VLH35_Angle_Bus {
   static void select_enc_(uint8_t id);
 };
 
+class KillSwitchRelay {
+public:
+  static bool dead;
+  static long kill_time;
+
+  static void init();
+  
+  //these two functions deal with the relay that kills all power to the motors
+  static void reset();
+  static void kill();
+
+  //the main loop, to be run before the effort values are assigned to motors
+  static void logic(RobotEffort &effort);
+
+  //sets an individual motor to 0% power.
+  static void kill_motor(int id, RobotEffort &effort);
+
+private:
+  //the pin to cut power to all motors. Active low to kill
+  static constexpr int kill_pin = 11;
+  static constexpr float drive_kill_curr = 7.0;
+  static constexpr float exdep_kill_curr = 25.0;
+
+  //the threshold at which the motor is set to 0% power 
+  static constexpr int cutoff_thresh = 1000;
+  //the threshold at which a motor set a 0% power is allowed to turn back on
+  static constexpr int reset_thresh = 500;
+
+  //Every cycle that a motor is overcurrent, a counter increases by this amount
+  static constexpr int cutoff_increase = 3;
```

Comment:
tad misleading. the decay runs every loop, even if there's an overcurrent, its effectively increasing by (cutoff_increase - cutoff_decay) during an overcurrent state

Reply:
made clearer